pkg/asset/store/store.go

```diff
@@ -53,18 +55,18 @@ type storeImpl struct {
 }
 
 // NewStore returns an asset store that implements the asset.Store interface.
-func NewStore(dir string) (asset.Store, error) {
-	return newStore(dir)
+func NewStore(dir string, validate bool) (asset.Store, error) {
+	return newStore(dir, validate)
 }
 
-func newStore(dir string) (*storeImpl, error) {
+func newStore(dir string, validate bool) (*storeImpl, error) {
```

Comment:
I see that you would like the validation to be run in the create path and not the destroy path. Can we have a separate validate() method that the top level function calls after they get a newStore()? validateStore() can then be called only in the create path.